src/applications/travel-pay/containers/TravelPayStatusApp.jsx

```diff
@@ -160,11 +160,11 @@ export default function App({ children }) {
 
   useEffect(
     () => {
-      if (userLoggedIn) {
+      if (userLoggedIn && travelClaims.length === 0) {
```

Comment:
Singe we're using React Router and getting the single-page application UI benefits of that, if someone navigates to a claim details page and _back_ to this page, the current state will already have claims and so another fetch would be redundant.

I don't think I'm missing any edge cases in that logic, but I'll double check and add a clarifying comment in either case

Reply:
I think I'm missing something, probably lacking React knowledge. My question was about the first time someone comes to this page - the first time the `useEffect` fires won't it check for `travelClaims.length === 0`? I can see that this is working locally, so maybe you can let me know what I'm missing offline

Reply:
> My question was about the first time someone comes to this page - the first time the useEffect fires won't it check for travelClaims.length === 0?

Yes, which is the only time we want claims fetching to fire.

> won't travelClaims always have a length of 0?

Client-side routing means this isn't always true. For instance, in this scenario, it wouldn't:

1. Claims status page loads. `travelClaims.length` is 0, so claims get fetch
2. User then navigates to a claim details page. Since right now this is just displaying info that's exclusively found in the claim data from the status page, there's no additional fetch for the claim details
3. User navigates from claim details **_back_** to claim status page using the breadcrumb navigation. Since this uses client-side routing, the previously fetched `travelClaims` are still in the state, and so re-fetching the claims is not needed